### ISSUE TITLE
[Topic RL] All components - display focus outline via keyboard only

### ISSUE THREAD

**Issue by ilhan007** [SAP/ui5-webcomponents#5752]
### **Feature Request Description**

As part of the Fiori Next initiative, along with the Horizon theme family, we would like to change the focus outline behaviour as follows: focus should be present only on pressing keyboard keys. Currently we apply focus upon mouse interaction (click) as well. So, we would have to distinguish between these two cases and apply focus outline only in the first case.

Origin issue: https://github.com/SAP/ui5-webcomponents/issues/3986

### **Proposed Solution**
There is a native solution, you can make use of [:focus-visible pseudo class](https://drafts.csswg.org/selectors/#the-focus-visible-pseudo) to set the focus only via keyboard navigation.

When switching to the CSS solution, the `focused` state and the event handlers that update it should be removed.

### **Additional Context**
For use-cases when the focus outline should be displayed on another HTML element other than the focused one, the ":focus-visible" won't work as it matches the actual focused element. 

### **Example**
```css
:host([focused]) .ui5-button-root:after {

}
will become
.ui5-button-root:focus-visible.ui5-button-root:after {

}

```

### Support
**Contact UI5 Web Components Core team for all cases that the CSS solution won't work to discuss how to 
handle them.**

### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

A clear and concise description of the impact/urgency of the required feature.

### **Stakeholder Info** (if applicable)

- Organization: SAP UX

**Comment by ilhan007** [comment]
Hello @SAP/ui5-webcomponents-topic-rl this is a new Feature Request that needs to be evaluated, and eventually planned.

Thanks,
ilhan

**Comment by ilhan007** [comment]
Closed in favor: https://github.com/SAP/ui5-webcomponents/issues/8250
